### PR TITLE
fix multi-channel.yaml + split ingress + podLabels + cleanup

### LINKED ISSUE #126: PWA: Multichannel source is fixed allowing no changes to `./nginx/multi-channel.yaml`
## Actual behavior ##

When making changes to pwa file `./nginx/multi-channel.yaml` they have no effect on the multichannel configuration of nginx. Everything stays in default.

## Expected behavior ##

When making changes to pwa file `./nginx/multi-channel.yaml` they become effective as no other multichannel configuration object is existing in the deployment.

### PR DESCRIPTION
**Pull request by shauke**
## PR Type

[x] Bugfix
[x] Feature
[x] Refactoring (no functional changes, no API changes)
[x] Documentation content changes

## What Is the New Behavior?
 * `multi-channel.yaml` fallback fix
 * split ingress
 * podLabels
 * cleaned up deprecated or not relevant code parts

Issue Number: Closes #126

## Release
The new functionality of this release will result in PWA Helm Chart version 0.7.0.

## Does this PR Introduce a Breaking Change?

[x] Yes

Cleaned up deprecated code that should not have been used anymore anyways.

### CODE REVIEW

File: charts/pwa/README.md [hunk 1]
```diff
@@ -35,10 +35,68 @@ spec:
...
+In addition the version changes and possibly necessary migration information is given.
+
+| Chart | PWA    | Changes                                                                                                                                                                                           | Migration Information                                                                                                   |
+| ----- | ------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------- |
+| 0.7.0 | 1.0.0  | re-enabled support for `multi-channel.yaml` and `caching-ignore-params.yaml` source code fallbacks<br/>add additional ingress for domain whitelisting<br/>add labels on deployment and pod levels | removed deprecated configuration options<br/>`cache.enabled` - was not optional<br/>`cache.channels`<br/>`upstream.icm` |
```
Comment (jometzner) [inline charts/pwa/README.md:88]: You think it's enough to mention the removal of deprecated customization options here? I have seen the usage of e.g. `upstream.icm` in a lot customer's flux configs. Maybe a migration note would be helpful?

File: charts/pwa/templates/deployment_cache.yaml [hunk 2]
```diff
@@ -42,23 +47,22 @@ spec:
           - name: UPSTREAM_PWA
             value: "http://{{ include "pwa-main.fullname" . }}:{{ .Values.service.port }}"
           - name: ICM_BASE_URL
-            value: "{{ coalesce .Values.upstream.icm .Values.upstream.icmBaseURL }}"
+            value: "{{ .Values.upstream.icmBaseURL }}"
+        {{- if .Values.cache.multiChannel }}
```
Comment (jometzner) [inline charts/pwa/templates/deployment_cache.yaml:51]: I still disagree to make this configuration optional. This is a deployment setting reflecting how NGInx is mapping external URLs to Application Server URLs. Making this part of the artifact being deployed sounds silly to me.

**Review by jometzner — COMMENTED** [review]
Have a look at my comments

File: charts/pwa/templates/deployment_cache.yaml (hunk 2 above)
Comment (jometzner) [reply charts/pwa/templates/deployment_cache.yaml:51]: Yes. With this change the implementation partner can unset the multi channel config of `values.yaml` (i.e. Helm) giving precedence to whatever contents of `nginx/multi-channel.yaml` (i.e. Source Code). As a result fixing #126 

I know this is open to debate. I also know that expectations from the implementation partner differ. They see the `multi-channel.yaml` as part of the Intershop PWA source code. So they use it to configure their URLs.

File: charts/pwa/README.md (hunk 1 above)
Comment (shauke) [reply charts/pwa/README.md]: I added migration notes for 0.7.0. With this everybody should be able to figure out what to do.

File: charts/pwa/templates/deployment_cache.yaml (hunk 2 above)
Comment (shauke) [reply charts/pwa/templates/deployment_cache.yaml:51]: Well I would say the change completely supports the way multi channel configuration should be done in your opinion while adding the also requested and used way of configuring it via project source code.
So there is no need for debate, the customer and partners can still use the "right" way. ;)

**Review by marschmidt89 — DISMISSED** [review]
Reviewed documentation files.

**Comment by shauke** [comment]
Thank you @jometzner and @marschmidt89 for reviewing.

**Review by jometzner — DISMISSED** [review]

**Review by andreassteinmann — DISMISSED** [review]

**Review by marschmidt89 — APPROVED** [review]
2nd documentation review done and approved.

**Review by hodor9 — APPROVED** [review]
Docu was adapted for ingresssplit and podLabels. All good.

**Pull request merged**